src/utils/source.js

```diff
@@ -9,6 +9,7 @@ import { isOriginalId } from "devtools-source-map";
 import { endTruncateStr } from "./utils";
 import { basename } from "../utils/path";
 import { parse as parseURL } from "url";
+import { isReactComponent } from "../workers/parser/frameworks";
```

Comment:
ooh, this call will have to be async because it requires parsing...

if we want to do this, we'll need to add it to another place, like on `loadSourceText`.
i'm not sure if we're already parsing 


```diff

+export function setSourceMetaData(sourceId: SourceId) {
+  return async ({ dispatch, getState }: ThunkArgs) => {
+    const sourceRecord = getSource(getState(), sourceId);
+    if (!sourceRecord) {
+      return;
+    }
+
+    const source = sourceRecord.toJS();
+    if (!source.text || source.isWasm || hasSymbols(getState(), source)) {
+      return;
+    }
+
+    const isReactComponent = await isReactComponent(source.get("id"));
+    const hasJSX = await hasJSX(source.get("id"));
+
+    // would update the source record...
+    dispatch({
+      type: "SOURCE_META_DATA",
+      isReactComponent,
+      hasJSX
+    });
+  };
+}
```

Reply:
@jasonLaster  I looked at https://github.com/devtools-html/debugger.html/blob/c5f08e7dd63c7fc0f238c70ca016aa474e831105/src/workers/parser/tests/framework.spec.js#L6
There is no indication that it is async. 
Also https://github.com/devtoolshtml/debugger.html/blob/master/src/workers/parser/frameworks.js doesn't look to me if it's async. What am I missing/overseeing?

Reply:
Everything in the worker is async. Also needs to be parsed